pkg/deploy/generator/resources_gateway.go

```diff
@@ -618,9 +618,8 @@ done
 							{
 								Name: to.StringPtr("gateway-vmss-nic"),
 								VirtualMachineScaleSetNetworkConfigurationProperties: &mgmtcompute.VirtualMachineScaleSetNetworkConfigurationProperties{
-									Primary: to.BoolPtr(true),
-									// TODO: enable this (requires >= Standard_D4s_v3)
-									// EnableAcceleratedNetworking: to.BoolPtr(true),
+									Primary:                     to.BoolPtr(true),
+									EnableAcceleratedNetworking: to.BoolPtr(true),
```

Comment:
You should be able to put in `[not(parameters('disableGatewayAcceleratedNetworking'))]` instead of replacing it in `templates.go` below.  

The ones referenced in `pkg/deploy/generator/templates.go` are a "hack" for working around specific types or structs that we don't have access to.  This shouldn't be a requirement for enabling accelerated networking.

Reply:
After some discussion, it's a bool pointer so it needs to be in `templates.go` to modify it as we can't assign the value to a string.